StarryExpanse/EverPresent/StrangerController.cpp

```diff
@@ -93,13 +93,44 @@ void AStrangerController::AddHorizontalMouseScan(float amount) {
   if (!IsCursorLockedToCenter) {
     HorizontalMousePosition =
         FMath::Clamp(HorizontalMousePosition + amount, 0.0f, 1.0f);
+
+    // check to see if we're within +tolerance for screen border
+    if (1 - HorizontalMousePosition < HorizontalMouseNudgeThreshold) 
+	{
+		//rotate the camera in accordance with how close the cursor is to the edge of the screen
+      this->AddYawInput(HorizontalMouseNudgeThreshold -
+                        (1 - HorizontalMousePosition));
+    }
+
+	// check to see if we're within -tolerance for screen border
+	if (HorizontalMousePosition < HorizontalMouseNudgeThreshold)
+	{
+      // rotate the camera in accordance with how close the cursor is to the edge of the screen
+      this->AddYawInput(
+              -(HorizontalMouseNudgeThreshold - HorizontalMousePosition));
+	}
   }
 }
 
 void AStrangerController::AddVerticalMouseScan(float amount) {
   if (!IsCursorLockedToCenter) {
     VerticalMousePosition =
         FMath::Clamp(VerticalMousePosition + amount, 0.0f, 1.0f);
+
+    // check to see if we're within +tolerance for screen border
+    if (1 - VerticalMousePosition < VerticalMouseNudgeThreshold) 
+	{
+      // rotate the camera in accordance with how close the cursor is to the edge of the screen
+      this->AddPitchInput(
+          VerticalMouseNudgeThreshold - (1 - VerticalMousePosition));
```

Comment:
Not certain this is the right expression. We would want the amount nudged to be proportional to how much of the threshold has been moved across. If the cursor has barely impinged into the threshold, let's say 1% of the threshold, then this pitch input argument would be VerticalMouseNudgeThreshold * 99%. If the cursor has impinged a lot (50%), the number would be less, VerticalMouseNudgeThreshold * 50%. Does that sound right? Either way, some more explicit variable names would make things a bit clearer. For instance
```
auto farImpingement = FMath::Clamp(1 - VerticalMousePosition, 0.0f, 1.0f);
auto nearImpingement = FMath::Clamp(VerticalMousePosition, 0.0f, 1.0f);
```

Reply:
in testing the current code appears to do just that - if the cursor is closer to the edge, the camera moves faster.  if nearer to the center, it moves slower.  
intent was to avoid making more variables/function calls than necessary.  
may be more of a style thing.  happy to use clamp instead if desired.

Reply:
i see what you're saying.  it's a bit counter-intuitive but knowing that the screen space in this function is measured from -1.0 to 1.0 it seemed to be the method that required the fewest calculations and function calls.  also, the commit last night allows for a multiplier so you can have a really small threshold and a really fast (or slow) turn rate.